Hiding input when the queue is full #DS-730 (#831)
lmc-eu/spirit-design-system: packages/web-react/src/components/FileUploader/README.md

</FileUploader>
```

> ### Dismissible Input

Author: The task is about dismissible, not about file queue

| `maxFileSize`      | `number`                             | 1000000 | ✕        | The maximum size of the uploaded file                                                                                                                           |
| `maxUploadedFiles` | `number`                             | 10      | ✕        | Maximum file upload queue size                                                                                                                                  |
| `isMultiple`       | `boolean`                            | -       | ✕        | When multiple files can be selected at once                                                                                                                     |

> | `isDismissible`    | `boolean`                            | -       | ✕        | When to hide FileUploaderInput after reaching file limit in `maxUploadedFiles`                                                                                  |

Reviewer: How about something more extensible:

```suggestion
| `queueLimitBehavior`    | `disable`, `hide` | `hide` | ✕        | Behavior of FileUploaderInput after reaching file limit in `maxUploadedFiles`                                                                                  |
```

1. We can add more options in the future.
2. IMO, [_dismiss_](https://slovnik.seznam.cz/preklad/anglicky_cesky/dismiss) is not the right word for this… 🤔

Reviewer 2: I have asked ChatGPT to suggest some other names, here are a few that I consider usable: `visibilityOnLimit`, `visibilityOnLimitReached`, `maxUploadedLimitBehavior`, `uploadLimitBehavior`, and `maxUploadedFilesAction`.

Author: The default state probably shouldn't be 'hidden'. On the contrary, there should be one more value, perhaps 'none', or better `undefined'. Because the user can not specify `queueLimitBehavior` at all and nothing will happen but throw errors. But if I set 'hide' or 'disable' it will either be hidden or disabled

Reviewer 2: What do you think about the default value for this should be `disabled`? Or it is correct that the actual state (meaning without any visual change after the limit is reached) is ok and then there should be the default value of `undefined`?

Author: I set there as default value 'none' for internal states, then it's like `undefined`

via `inputProps` and `listProps`.
`UncontrolledFileUploaderBaseProps` extends `FileUploaderBaseProps`.

| Prop name             | Type                                    | Default | Required | Description                                                                    |
| --------------------- | --------------------------------------- | ------- | -------- | ------------------------------------------------------------------------------ |
| `id`                  | `string`                                | -       | ✔        | UncontrolledFileUploader id                                                    |
| `inputId`             | `string`                                | -       | ✔        | FileUploaderInput id                                                           |
| `inputLabel`          | `string`                                | -       | ✔        | FileUploaderInput label                                                        |
| `inputName`           | `string`                                | -       | ✔        | FileUploaderInput name                                                         |

> | `inputProps`          | `Partial<FileUploaderInputBaseProps>`   | -       | ✕        | Rest of FileUploaderInput props                                                |

Reviewer: Oh, now I noticed 🙈. When a props is not required, it has a default value (`''`, `null`, `false`, … see the [`Button` `README`](https://github.com/lmc-eu/spirit-design-system/blob/main/packages/web-react/src/components/Button/README.md#available-props), for example) that needs to be listed here in the API table.

Now when we are on it, if you have time, would please fix it? 🙏🏻  But we can do it later, no pressure.

cc @literat

Author: But that's wrong. Not always is default value `null` or `false`. Depends on type. For example, the unspecified value for children is not `null` but `undefined`
